libs/SalesforceSDKCore/SalesforceSDKCore/Classes/OAuth/SFOAuthCredentials.m

```diff
@@ -312,5 +313,53 @@ - (BOOL)hasPropertyValueChangedForKey:(NSString *) key {
     return [_credentialsChangeSet objectForKey:key]!=nil;
 }
 
+- (NSURL *)overrideDomainIfNeeded{
+    
+    NSString *refreshDomain = self.communityId ? self.communityUrl.absoluteString : self.domain;
```

Comment:
This seems incorrect. Refresh domain should always go against the configured login host, i.e. login URL. This is because the token endpoint is a part of the login pool and is agnostic of instance URL and/or community URL. This is especially important when pods are taken down and/or migrated to a different instance server.

Reply:
@bhariharan The code has been remained the same since 5.0.  https://github.com/forcedotcom/SalesforceMobileSDK-iOS/blob/9024d4ccc40b770e1d8cdb1fddd9b489b30b6392/libs/SalesforceSDKCore/SalesforceSDKCore/Classes/OAuth/SFOAuthCoordinator.m#L647

Reply:
Right - we can take a look separately after this PR.